okhttp/src/main/java/okhttp3/internal/framed/Hpack.java

```diff
@@ -367,27 +368,125 @@ ByteString readByteString() throws IOException {
   }
 
   static final class Writer {
+    private static final byte SEPARATED_TOKEN = ':';
+    private static final int SETTINGS_HEADER_TABLE_SIZE = 4096;
+
     private final Buffer out;
+    private final Map<ByteString, Integer> headerStringToDynamicIndex =
+        new HashMap<ByteString, Integer>();
```

Comment:
nit: can you please do a LinkedHashMap? Nondeterministic code is one of my pet peeves.